### ISSUE TITLE
Can't trigger events in Bro_RPC.events.bif.bro

### ISSUE THREAD

**Issue by TheGodOfObelisk** [zeek/zeek#684]
Dear developers.
I am very sorry to disturb you. I use the old version of Zeek(Bro 2.x). And I use bro to analyze the data set [LLDoS1.0(phase-2-dump)](https://archive.ll.mit.edu/ideval/data/2000/LLS_DDOS_1.0.html), in which there are many RPC calls. But none of the events related to RPC are triggered. 
Such as this:
`event rpc_call(c: connection, xid: count, prog: count, ver: count, proc: count, call_len: count){
    print "rpc_call!";
}`
Are all of these RPC events implemented? Or there is something wrong with the codes? Thank you very much!

**Comment by jsiwek** [comment]
The RPC-based protocol analyzers (portmapper, nfs, mount) aren't enabled by default as mentioned in the docs for the events.  You could enable, for example, the portmapper analyzer in a script like this (or use `bro_init` instead of `zeek_init` for older versions):

```zeek
const pm_ports = { 111/udp };
redef likely_server_ports += { pm_ports };

event zeek_init()
	{
	Analyzer::register_for_ports(Analyzer::ANALYZER_PORTMAPPER, pm_ports);
	}
```

Others would need different port definitions to enable.  But also looking at some of the pcaps you referenced and trying that, it seems like the RPC parser/analyzer is just not working correctly (at a glance the parsing logic in `RPC_CallInfo::RPC_CallInfo()` in `RPC.cc` seems to fail).  Part of the reason these analyzer are disabled by default is they were never well-maintained or tested, but it may not be far of from getting it to work if you were interested in taking a look.  Any bug fixes you figure out would be appreciated.

**Comment by TheGodOfObelisk** [comment]
> The RPC-based protocol analyzers (portmapper, nfs, mount) aren't enabled by default as mentioned in the docs for the events. You could enable, for example, the portmapper analyzer in a script like this (or use `bro_init` instead of `zeek_init` for older versions):
> 
> ```zeek
> const pm_ports = { 111/udp };
> redef likely_server_ports += { pm_ports };
> 
> event zeek_init()
> 	{
> 	Analyzer::register_for_ports(Analyzer::ANALYZER_PORTMAPPER, pm_ports);
> 	}
> ```
> 
> Others would need different port definitions to enable. But also looking at some of the pcaps you referenced and trying that, it seems like the RPC parser/analyzer is just not working correctly (at a glance the parsing logic in `RPC_CallInfo::RPC_CallInfo()` in `RPC.cc` seems to fail). Part of the reason these analyzer are disabled by default is they were never well-maintained or tested, but it may not be far of from getting it to work if you were interested in taking a look. Any bug fixes you figure out would be appreciated.

Thank you for your suggestions! I tried the in-built analyzer called ANALYZER_CONTENTS_RPC and part of the code:
`
const pm_ports = { 111/udp };
redef likely_server_ports += {pm_ports};

event zeek_init(){
    # enable RPC-based protocol analyzers
    Analyzer::register_for_ports(Analyzer::ANALYZER_CONTENTS_RPC, pm_ports);
    Analyzer::enable_analyzer(Analyzer::ANALYZER_CONTENTS_RPC);
}

event rpc_call(c: connection, xid: count, prog: count, ver: count, proc: count, call_len: count){ 
    print "rpc_call!"; 
}
# ...
`
After executing the script, I got these warning messages.
`
952441627.223333 internal warning: analyzer CONTENTS_RPC cannot be instantiated dynamically
952441687.346962 internal warning: analyzer CONTENTS_RPC cannot be instantiated dynamically
952441687.488421 internal warning: analyzer CONTENTS_RPC cannot be instantiated dynamically
952441807.738458 internal warning: analyzer CONTENTS_RPC cannot be instantiated dynamically
952441867.860918 internal warning: analyzer CONTENTS_RPC cannot be instantiated dynamically
952441868.346402 internal warning: analyzer CONTENTS_RPC cannot be instantiated dynamically
952441868.930192 internal warning: analyzer CONTENTS_RPC cannot be instantiated dynamically
952441869.493558 internal warning: analyzer CONTENTS_RPC cannot be instantiated dynamically
`
It seems that the number of warnings equals to the number of RPC packets in data set. I think it goes into event rpc_call successfully but something unexpected happened. Does it have anything to do with the potential bug in RPC.cc? 
I am sorry that I am not capable of fixing the bug in zeek's source code by myself. May I have some more hints? Thank you!

**Comment by TheGodOfObelisk** [comment]
I saw these statements under RPC-based events' doc.

**Zeek’s current default configuration does not activate the protocol analyzer that generates this event; the corresponding script has not yet been ported. To still enable this event, one needs to register a port for it or add a DPD payload signature.**

Does adding a DPD payload work? If so, how can I add a DPD payload signature? I have loaded the scripts in path <prefix>/share/zeek/policy/frameworks/dpd/.
Thank you.

**Comment by jsiwek** [comment]
* Adding a DPD payload won't change anything
* The `CONTENTS_RPC` analyzer isn't meant to be added directly in user scripts, it's something that gets used internally only.  The only available RPC-based analyzers are `PORTMAPPER`, `NFS`, and `MOUNT`.  If you need other ones, then you may have to look into writing a plugin that does what you need.
* The underlying bug your encountering I think should be fixed via #685 if you want to give that branch a try.

**Comment by TheGodOfObelisk** [comment]
Thanks!
